docs/reference/search/retriever_examples.asciidoc

```diff
@@ -0,0 +1,416 @@
+[[retriever_examples]]
```

Comment:
```suggestion
[[retriever_examples]]
==== Retrievers examples
```

Need this now that you're including it and not using tags.

You'll need to deepen the subheadings accordingly. Also consider giving the subheadings unique IDs too for likability :)